oneview_redfish_toolkit/api/manager.py

```diff
@@ -27,31 +32,54 @@ class Manager(RedfishJsonValidator):
 
     SCHEMA_NAME = 'Manager'
 
-    def __init__(self, oneview_resource, firmware_version):
+    def __init__(self, oneview_appliance_info, oneview_appliance_state,
+                 oneview_appliance_health_status):
         """Manager constructor
 
             Populates self.redfish with some hardcoded Manager
             values and with the response from OneView.
 
             Args:
-                oneview_resource: A dict for ServerHardware or Enclosure
+                oneview_appliance_info: An Oneview's appliance info dict
+                oneview_appliance_state: An Oneview's appliance status dict
+                oneview_appliance_health_status: An Oneview's appliance
+                health state dict
         """
 
         super().__init__(self.SCHEMA_NAME)
 
         self.redfish["@odata.type"] = self.get_odata_type()
-        self.redfish["Id"] = oneview_resource['uuid']
-        self.redfish["Description"] = None
-        self.redfish["FirmwareVersion"] = firmware_version
-
-        # TODO(victorhugorodrigues): update state and health properties
-        # self.redfish["Status"] = collections.OrderedDict()
-        # state, _ = status_mapping.\
-        #     get_redfish_server_hardware_status_struct(oneview_resource)
-        # self.redfish["Status"]["State"] = state
-        # self.redfish["Status"]["Health"] = health
-
+        self.redfish["Id"] = oneview_appliance_info['uuid']
+        self.redfish["Description"] = oneview_appliance_info["family"]
+        self.redfish["ManagerType"] = "Service"
+        self.redfish["FirmwareVersion"] = \
+            oneview_appliance_info["softwareVersion"]
+        self.redfish["Status"] = collections.OrderedDict()
+        state = status_mapping.APPLIANCE_STATE_TO_REDFISH_STATE.\
+            get(oneview_appliance_state["state"])
+        health = self._get_highest_health_state(
+            oneview_appliance_health_status["members"])
+        self.redfish["Status"]["State"] = state
+        self.redfish["Status"]["Health"] = health
         self.redfish["@odata.context"] = \
             "/redfish/v1/$metadata#Manager.Manager"
         self.redfish["@odata.id"] = \
-            "/redfish/v1/Managers/" + oneview_resource['uuid']
+            "/redfish/v1/Managers/" + oneview_appliance_info['uuid']
+
+    @staticmethod
+    def _get_highest_health_state(health_state_members):
```

Comment:
I have a suggestion, let me know if it works well for this situation and what is your opinion:
```python
def _get_highest_health_state(health_state_members):
    health_state_result = None
    highest_status = 0
    
    for member in health_state_members:
        redfish_health_state = status_mapping.MANAGER_HEALTH_STATE.\
                get(member["severity"])
        current_status = status_mapping.CRITICALITY_STATUS[redfish_health_state]

        if current_status > highest_status:
            highest_status = current_status
            health_state_result = redfish_health_state
    
    return health_state_result
```

It is because we can leverage the "for loop" to get the result desired.

Reply:
I rollback this code to use lambda again, because the "for" present on lambda is more elegant than the way you suggest. It avoids to create more code and simplify this into one line :+1: 

But I appreciate your suggestion :)

Let's see what @adriane-cardozo and @victorhugorodrigues think about this.

Reply:
The way @ricardogpsf suggested is a more explicit way and for me easier to read. But the way you did is a more elegant way using some fancy code. In terms of performance, the way you did the computer execution will have to iterate twice to reach the result, meanwhile the way Ricardo proposed will just iterate once, so I would choose the the way Ricardo proposed.